Feat: 회원 로그인 기능 구현중

gyhcom opened the pull request:
1.Login관련 Process,Request,Command 추가
2.인증 구현(email로 인증)
3.토큰방식 인가 진행예정

Commit:
Feat: Add Member signUp api

Commit:
Feat: Add Member signUp api
개행 정리

Commit:
main gradle starlucks-member add

gyhcom committed:
Feat: Add Member signUp api (#1)

개행 정리

Co-authored-by: gyhcom <gyeonho@gamil.com>

gyhcom committed:
f-lab -> gyhcom (#2)

* Feat: Add Member signUp api (#16)

* Feat: Add Member signUp api

* Feat: Add Member signUp api
개행 정리

---------

Co-authored-by: gyhcom <gyeonho@gamil.com>

* #19 Add mysql configuration (#20)

* Add docker-compose

---------

Co-authored-by: gyhcom <gyeonho@gamil.com>
Co-authored-by: Jisu Na <kuu2002@naver.com>

Commit:
Feat: 회원 가입 JPA 기능 구현
기존 InMemory -> Jpa 기능변경
Common ErrorCode, ApiResponse 추가
Api Test 완료
테이블 구조 수정 필요

gyhcom committed:
Merge branch 'main' into main

Commit:
Merge branch 'MemberSignup_Jpa'

Commit:
Feat: 회원 가입 JPA 에러수정

Feat: 회원 가입 JPA 에러수정
checkstyle 에러 수정

Commit:
FEAT:Code review 수정
DDL 소문자로 수정 및 unique 추가
InMemory,Jpa 패키지 추가,
Member Entity @Table 추가,
memberTable.ddl 삭제

Commit:
FEAT:패스워드 암호화

Commit:
FEAT:InMemory 삭제& Sout 삭제

FEAT:InMemory 삭제& Sout 삭제

gyhcom committed:
Merge branch 'f-lab-edu:main' into main

Commit:
Feat: 회원 로그인 기능 구현중
1.Login관련 Process,Request,Command 추가
2.인증 구현(email로 인증)
3.토큰방식 인가 진행예정

MemberPassWordEncoder 추가

Commit:
Feat: 회원 로그인 기능 구현중
1.Redis 연동 완료
2.로그인시 캐시저장

Feat: 회원 로그인 기능 구현중
1.Member Role 추가 및 DB Column Role 추가

gyhcom commented on starlucks-member/src/main/java/com/starlucks/member/infrastructure/persistence/redis/RedisMemberRepository.java:
```diff
@@ -0,0 +1,35 @@
+
+    public void save(AuthMember authMember) {
+        System.out.println("authmember_id =" + authMember.getAuthId());
+        System.out.println("authmember =" + authMember);
```
로그는 삭제하도록 하겠습니다.

JisuNa commented on docker/apm-docker-compose.yml:
```diff
@@ -0,0 +1,46 @@
+version: '3.7'
+
+services:
+  elasticsearch:
+    image: docker.elastic.co/elasticsearch/elasticsearch:7.15.0
+    container_name: starlucks_elastic
```
APM 설정은 별도의 PR로 됐으면 구분하기 좋았을 것 같아요.

JisuNa commented on starlucks-main/src/main/resources/db/migration/V1.1__user_role_ddl.sql:
```diff
@@ -0,0 +1,2 @@
+-- Member Role 추가
+alter table member ADD COLUMN role varchar(100) not null comment '권한';
```
권한에 데이터가 길이가 큰 경우가 있을까요??

gyhcom replied:
긴 경우는 없을꺼 같습니다. varchar(20) 정도로 수정하겠습니다.

JisuNa commented on starlucks-member/src/main/java/com/starlucks/member/application/processor/MemberLoginProcessor.java:
```diff
@@ -0,0 +1,54 @@
+        var member = memberRepository.findByEmail(command.getEmail());
+
+        if (member == null) {
+            throw new NullPointerException("잘못된 회원입니다.");
```
Custom Exception으로 정의하는게 좋을 것 같습니다.

gyhcom replied:
Exception 따로 만들어서 지정하겠습니다.

JisuNa commented on starlucks-member/src/main/java/com/starlucks/member/domain/entity/Role.java:
```diff
@@ -0,0 +1,7 @@
+package com.starlucks.member.domain.entity;
+
+public enum Role {
+    USER,
+    MANAGER,
+    ADMIN,
```
`Member` 모듈은 일반 사용자로 판단되는데, `MANAGER`와 `ADMIN`권한이 없어도 될 것 같습니다.
일반 사용자의 권한 구분이 필요한 경우가 있을까요??

JisuNa commented on starlucks-member/src/main/java/com/starlucks/member/infrastructure/MemberRepositoryAdapter.java:
```diff
@@ -17,4 +18,10 @@ public MemberRepositoryAdapter(JpaMemberRepository jpaMemberRepository) {
+    @Override
+    public Member findByEmail(String email) {
+        return this.jpaMemberRepository.findByEmail(email)
+            .orElseThrow(()-> new NullPointerException("가입되지 않은 회원입니다."));
```
Custom Exception으로 정의하는게 좋을 것 같습니다.

gyhcom replied:
네넵 알겠습니다! :)

JisuNa commented on starlucks-member/src/main/java/com/starlucks/member/infrastructure/config/RedisConfig.java:
```diff
@@ -0,0 +1,33 @@
+
+@Configuration
+public class RedisConfig {
+    @Value("localhost")
+    //@Value("${spring.redis.session.host}")
+    private String redisHost;
+    @Value("16379")
+    //@Value("${spring.redis.session.port}")
+    private int redisPort;
```
주석되어 있는 것처럼 config를 읽어오는 게 좋을 것 같아요.

gyhcom replied:
네넵 알겠습니다! :)

JisuNa commented on starlucks-member/src/main/java/com/starlucks/member/infrastructure/properties/TokenProperties.java:
```diff
@@ -0,0 +1,8 @@
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class TokenProperties {
+
+}
```
사용되는 클래스인가요??

JisuNa commented on starlucks-member/src/main/java/com/starlucks/member/infrastructure/membersecurity/MemberPassWordEncoder.java:
```diff
@@ -0,0 +1,18 @@
+    }
+
+    @Override
+    public String encrypt(String password) {
+        return encoder.encode(password);
+    }
```
사용된 암호화 방식에 대해 설명 해주실 수 있을까요??

gyhcom replied:
SpringSecurity 에서 지원하는`PasswordEncoder` 인터페이스를 사용하였고   `BcryptPassEncoder` 구현 클래스를 통해 사용자의 비밀번호를 단방향 암호화로 진행했습니다.

```java
암호화 메소드 -> encoder()
비밀번호 비교 메소드 -> matches()
패스워드 안전점검 -> upgradeEncoding()
```

JisuNa commented:
추가적으로 `member` 테이블 생성을 다시보니 `id INT(10)`은 조금 작지 않을까 싶네요.
`id BIGINT UNSIGNED NOT NULL AUTO_INCREMENT UNIQUE` 변경이 되면 좋을 것 같습니다.

Commit:
Feat: 회원 로그인 기능 구현중
1.RedisConfig 수정
2.MemberException 생성

Commit:
Feat: 회원 로그인 기능 구현중
    1.Logout 기능 생성

Commit:
Feat: 회원 로그인 기능 구현중
    1.Password Matched 조건 추가
    2.@JisuNa 요구사항 변경

Commit:
Feat: 회원 중복메일로직 추가

gyhcom committed:
Merge branch 'f-lab-edu:main' into main

Commit:
Feat: LogFilter 삭제 MDC Filter 사용

Commit:
Feat: @Jisuna 요구사항
1. 필요없는 권한 삭제(Admin,Manager)
2. 필요 없는 파일 삭제(TokenProperties)

Feat: flaway 버전 충돌
1. user_role와 create_order 같은 버전으로
   1.1로 겹처 user_role을 1.0버전에다가 통합

RENAME : init_ddl -> create_member_ddl
